x-pack/solutions/observability/plugins/streams/server/lib/streams/client.ts

```diff
@@ -344,6 +345,47 @@ export class StreamsClient {
     };
   }
 
+  private async assertNoHierarchicalConflicts(definitionName: string) {
+    const streamNames = [...getAncestors(definitionName), definitionName];
+    const hasConflict = await Promise.all(
+      streamNames.map((streamName) => this.isStreamNameTaken(streamName))
```

Comment:
Small note: If we used arrow functions to define our methods we could write this simply as `streamNames.map(this.isStreamNameTaken)` since the context would already be bound.

Reply:
I'm always in favour of referencing functions vs inline callbacks, I wouldn't mind if you change the method declaration to an arrow function.